File: src/hole 26 - Morse Decoder/solution.java
/*
as of 01/04/2024 ranked 8th in bytes and 9th in chrs
*/
class M{public static void main(String[]a){String s="..ETINAMSDRGUKWOHBLZFCP.VX.Q.YJ.56.7...8.......94.......3...2.10";for(String w :a[0].split("          ")){for(String c:w.split("   ")){System.out.print(s.charAt(Integer.parseInt('1'+new StringBuffer(c.replace("▄▄▄","1").replace("▄","0").replace(" ","")).reverse().toString(),2)));}System.out.print(" ");}}}

File: src/hole 26 - Morse Decoder/solution.py
"""
see the generation of string s in magic-s-generate.py
as of 01/04/2023 solution ranked 61st in bytes and 74th in chrs.
"""
import sys
s="..ETINAMSDRGUKWOHBLZFCP.VX.Q.YJ.56.7...8.......94.......3...2.10";r=str.replace
for w in sys.argv[1].split(" "*10):
	for c in w.split(" "*3):print(s[int("1"+r(r(r(c,"▄"*3,"1"),"▄","0")," ","")[::-1],2)],end="")
	print(" ",end="")
